faust_large_message_serializer/config.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional, Callable, Dict, Tuple, Union
+from urllib.parse import urlparse
+
+from dataclasses import dataclass
+
+
+@dataclass
+class URIParser:
+    base_path: str
+
+    def __post_init__(self):
+        self.base_path = self.base_path.lower()
+
+    def parse_uri(self) -> Tuple[str, str, str]:
+        result = urlparse(self.base_path)
+        scheme = result.scheme
+        bucket = result.netloc
+        if len(result.path) != 0 and result.path[0] == "/":
+            path = result.path[1:]
+        else:
+            path = result.path
+        return scheme, bucket, path
+
+    def __repr__(self):
+        return self.base_path
+
+
+@dataclass
+class LargeMessageSerializerConfig:
+    base_path: Union[str, URIParser]
+    max_size: int
+    large_message_s3_secret_key: Optional[str] = None
+    large_message_s3_access_key: Optional[str] = None
+    large_message_s3_region: Optional[str] = None
+    large_message_s3_endpoint: Optional[str] = None
+    large_message_abs_connection_string: Optional[str] = None
+    large_message_blob_storage_custom_config: Optional[
+        Callable[[Dict[str, Optional[str]]], None]
+    ] = None
+
+    def __post_init__(self):
+        if not isinstance(self.base_path, str) and not isinstance(
+            self.base_path, URIParser
+        ):
+            raise ValueError("base_path should be an str or a URIParser")
+        if self.base_path is None:
+            raise ValueError("You need to give a base_path")
```

Comment:
In java we only lazily check that when storing or retrieving a message

Reply:
I mean ... I think here is a good place to do this. base_path is part of config and it doesn't have a default value, so you have to give it to the constructor of the LargeMessageSerializerConfig.

Reply:
But as long as all messages are small enough, the serializer would work fine then. That way, you can also have it as a filter for large messages because it would only then throw an exception if a message exceeds the size

Reply:
I am lost, did you mark the correct part ? How does the base_path relate to max_size ?

Reply:
The base_path will only be used once a message needs to be stored on S3/Azure because it exceeds the max size. Therefore, it won't be used if all messages are smaller. Thus, we should defer the check for base path to the point where it is actually used

Reply:
Here we check it in Java https://github.com/bakdata/kafka-large-message-serde/blob/master/large-message-core/src/main/java/com/bakdata/kafka/LargeMessageStoringClient.java#L107